packages/report-server/src/routes/fetchReport.ts

```diff
@@ -24,15 +24,30 @@ class FetchReportRouteHandler {
       throw new Error(`No report found with code ${params.reportCode}`);
     }
 
-    const orgUnit = await models.entity.findOne({ code: query.organisationUnitCode });
-    if (!orgUnit) {
-      throw new Error(`No entity found with code ${query.organisationUnitCode}`);
-    }
     const permissionGroup = await models.permissionGroup.findById(report.permission_group_id);
+    const orgUnitCodes = query.organisationUnitCodes.split(',');
+    const orgUnitsAndCodes = await Promise.all(
+      orgUnitCodes.map(async orgUnitCode => ({
+        orgUnitCode,
+        orgUnit: await models.entity.findOne({ code: orgUnitCode }),
```

Comment:
Interesting structure! Another option would be creating an `orgUnitByCode` dictionary, and check that all org units in the input are a key in it

Reply:
Yeah it is a bit funny, I mainly did it so that I could play nicely with `Promise.all()`

Tbh I still haven't found a way of handling loops with async calls in a way that I'm happy with 😝 

I might leave things how they are for now? If you have a clean example of how to do this with an `orgUnitByCode` dictionary tho I would happily take it 😄

Reply:
```js
import { reduceToSet } from '@tupaia/utils';

const requestedOrgUnitCodes = query.organisationUnitCodes.split(',');
const orgUnits = await models.entity.find({ code: orgUnitCodes });

const resultOrgUnitCodes = reduceToSet(orgUnits, 'code');
requestedOrgUnitCodes.forEach(code => {
  if (!resultOrgUnitCodes.has(code)) {
    throw new Error(`No entity found with code ${invalidOrgUnit.orgUnitCode}`);
  }
});
```

Actually, could you change that in a new PR cause this reduces multiple DB queries to one @rohan-bes ?

Reply:
Ah ofcourse! Why do i always forget you can return multiple db entries with one call 🤦 

Will do this 👍